python/lsst/meas/astrom/matcher_probabilistic.py

```diff
@@ -522,27 +517,39 @@ def match(
 
         n_ref_select = len(ref.extras.indices)
 
-        match_dist_max = config.match_dist_max
         coords_spherical = config.coord_format.coords_spherical
-        if coords_spherical:
-            match_dist_max = np.radians(match_dist_max / 3600.)
-
-        # Convert ra/dec sky coordinates to spherical vectors for accurate distances
-        func_convert = _radec_to_xyz if coords_spherical else np.vstack
-        vec_ref, vec_target = (
-            func_convert(cat.coord1[cat.extras.select], cat.coord2[cat.extras.select])
+        coords_ref, coords_target = (
+            (cat.coord1[cat.extras.select], cat.coord2[cat.extras.select])
             for cat in (ref, target)
         )
 
         # Generate K-d tree to compute distances
         logger.info('Generating cKDTree with match_n_max=%d', config.match_n_max)
-        tree_obj = cKDTree(vec_target)
 
-        scores, idxs_target_select = tree_obj.query(
-            vec_ref,
-            distance_upper_bound=match_dist_max,
-            k=config.match_n_max,
-        )
+        if coords_spherical:
+            match_dist_max = config.match_dist_max/3600.
+            with Matcher(coords_target[0], coords_target[1]) as matcher:
+                idxs_target_select = matcher.query_knn(
+                    coords_ref[0], coords_ref[1],
+                    distance_upper_bound=match_dist_max,
+                    k=config.match_n_max,
+                )
+        # Call scipy for non-spherical case
+        # The spherical case won't trigger, but the implementation is left for comparison, if needed
```

Comment:
Do you mean "The **non-spherical** case won't trigger"?

Reply:
How do you anticipate such comparisons becoming necessary?

Reply:
No, I mean that the spherical case won't use scipy but will use smatch instead. So the `_radec_to_xyz if coords_spherical else np.vstack` on line 542 will never actually call `_radec_to_xyz`.

I anticipate most users will match on spherical coordinates and so the else block won't get called much.